spec/features/post_index_spec.rb

```diff
@@ -0,0 +1,81 @@
+require 'rails_helper'
+
+RSpec.feature 'User Post Index Page', type: :feature do
+  let!(:user) { create(:user, :with_posts) }
+
+  scenario 'I can see the user\'s profile picture' do
+    visit user_posts_path(user)
+
+    expect(page).to have_selector("img[src$='#{user.photo}']")
+  end
+
+  scenario 'I can see the user\'s username' do
+    visit user_posts_path(user)
+
+    expect(page).to have_content(user.name)
+  end
+
+  scenario 'I can see the number of posts the user has written' do
+    visit user_posts_path(user)
+
+    expect(page).to have_content("Number of posts: #{user.posts_counter}")
+  end
+
+  scenario 'I can see a post\'s title' do
+    post = user.posts.first
+    visit user_posts_path(user)
+
+    expect(page).to have_content(post.id)
+  end
```

Comment:
- [ ] Good job on the test cases! However, there's a small adjustment needed in this particular test case. You mentioned `scenario 'I can see a post\'s title'`, but in the expectation, you are checking for the post ID. According to the project requirement, it should check for the post title. I recommend updating your code to verify the post title instead.

For example:
```ruby
expect(page).to have_content(@post.title)
```

This ensures that the test accurately reflects the project requirements. Keep up the excellent work! 🚀🧪

Reply:
Really!

Reply:
There was a typo in the index.html.erb file. Fixed. 😄